### ISSUE TITLE
two edf files of same subject

### ISSUE THREAD

**Issue by garvin2506** [jashubbard/itrackR#11]
Hi,

sorry for bothering you again, it's more a question than an issue this time: 
I have two separate edf & beh files for each subject: same ID, but two different task conditions that were performed in separate runs; the TrialIDs are unique across runs (1:40 for the first edf, 41:80 for the second).

I'm wondering what's the best way to do these analyses in your current framework (sticking closely to your nice vignette). Not surprisingly, treating them as two different subjects leads to a mix-up in the merging process because of the shared ID.

I would be grateful for any ideas!

**Comment by jashubbard** [comment]
If your goal is to produce separate timecourses (fixations or pupil) for the different sessions, then the most straightforward approach would be to create a `session` variable in your behavioral file before adding it. Then, when you produce your plots, you can include `session` as one of the grouping variables (just like `Task` or `Conflict` in the vignettes). Since you have different trial numbers for the sessions, then it's pretty straightforward: 

```
beh$session <- 1
beh$session[beh$Trial>40] <- 2
z <- add_behdata(z,beh)
```

This way, all the averages will be computed within-session and within-subject. Will this accomplish what you need?

**Comment by garvin2506** [comment]
not quite, because the problem starts already when I run set_index and find_messages:

   eyetrial starttime endtime duration    ID first_sample Results TrialID
      <dbl>     <dbl>   <dbl>    <dbl> <dbl>        <dbl>   <dbl>   <dbl>
1         1         1    9464     9463  3207     12609644    6463      42
2         1         1    9464     9463  3207     12609644    6463       1
3         1         1    9464     9463  3207     12609644 -795824      42
4         1         1    9464     9463  3207     12609644 -795824       1
5         2      9509   19002     9493  3207     12609644   15999      43
6         2      9509   19002     9493  3207     12609644   15999       2
7         2      9509   19002     9493  3207     12609644 -786288      43
8         2      9509   19002     9493  3207     12609644 -786288       2

Here are sample data with 2 edf files and one .rds file (the combined behavioral data, as in your vignette) of one subject: https://www.dropbox.com/sh/woigudoexzxya5u/AAALFWIw77jpdTTVm8xXRWoja?dl=0

here's the relevant code: 
z<- set_index(z,varnames=c('TrialID'), patterns=c('Fixation TrialID[1-84]*'), numeric.only=T)
z <- find_messages(z,varnames=c('Results'), patterns=c('^Results'), timestamp=T)
beh$session <- 1
beh$session[beh$TrialID>39] <- 2
z <- add_behdata(z,bet)
